### ISSUE TITLE
buildah bud issue

### ISSUE THREAD

**Issue by coderwangke** [containers/buildah#1112]
**Description**

**Steps to reproduce the issue:**
1. buildah bud -f Dockerfile -t xxxx:latest .

**Describe the results you received:**
STEP 1: FROM golang:1.10-alpine
Getting image source signatures
Copying blob sha256:4fe2ade4980c2dda4fc95858ebb981489baec8c1e4bd282ab1c3560be8ff9bde

 0 B / 2.10 MiB 
 42.70 KiB / 2.10 MiB 
 126.76 KiB / 2.10 MiB 
 286.76 KiB / 2.10 MiB 
 670.76 KiB / 2.10 MiB 
 1.50 MiB / 2.10 MiB 
 2.10 MiB / 2.10 MiB  1s
Copying blob sha256:2e793f0ebe8a0d4496ef2fca6d5d45493df6dc40503657b357e153fa91548171
STEP 2: MAINTAINER xxx <xxx@xxx.com>
STEP 3: WORKDIR /go/src/component-docker
STEP 4: COPY ./ /go/src/component-docker
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0xa2b835]

**Describe the results you expected:**
i guess this, 'from' is not success, so 'copy' get error, maybe this is a network question resulting in ‘from’ error. should ‘from’ return a error, not 'copy' return a error

**Output of `rpm -q buildah` or `apt list buildah`:**
buildah version 1.3 (image-spec 1.0.0, runtime-spec 1.0.0)

**Comment by coderwangke** [comment]
@rhatdan

**Comment by TomSweeneyRedHat** [comment]
Thanks for the issue report @coderwangke .  This maybe the same as #1105, but we'll take a look.

**Comment by rhatdan** [comment]
@coderwangke Why do you think the from was unsuccessful?

**Comment by rhatdan** [comment]
@coderwangke Is this repeatable, and could you do this with a --debug flag set on the bud command.

buildah --debug bud ...

**Comment by TomSweeneyRedHat** [comment]
@coderwangke I gave this a go with the latest from GitHub and it seemed to work for me.  I ran as root and ran it from a subdirectory rather than the top level directory.  Can you tell me if you ran it as root and from what directory?  Is my Dockerfile and commands below OK?  If you can, please upgrade and retest, it may be something that was fixed in v1.4.  Also as @rhatdan asked, a run with --debug would be helpful too.

```
# buildah version
Version:         1.5-dev
Go Version:      go1.9.7
Image Spec:      1.0.0
Runtime Spec:    1.0.0
CNI Spec:        0.4.0
libcni Version:  v0.7.0-alpha1
Git Commit:      52d797d6
Built:           Tue Oct 16 19:56:07 2018
OS/Arch:         linux/amd64

# cat Dockerfile
FROM golang:1.10-alpine
MAINTAINER tom tsweeney@redhat.com
WORKDIR /go/src/component-docker
COPY ./ /go/src/component-docker

# buildah bud -f Dockerfile -t tom:latest .
STEP 1: FROM golang:1.10-alpine
Getting image source signatures
Copying blob sha256:4fe2ade4980c2dda4fc95858ebb981489baec8c1e4bd282ab1c3560be8ff9bde
 2.10 MiB / 2.10 MiB [======================================================] 0s
Copying blob sha256:2e793f0ebe8a0d4496ef2fca6d5d45493df6dc40503657b357e153fa91548171
 301.73 KiB / 301.73 KiB [==================================================] 0s
Copying blob sha256:77995fba19184aad4b702852da2d04d7b92693f2923499f5d388fd8d2ce2dd81
 153 B / 153 B [============================================================] 0s
Copying blob sha256:4495499e856d8d6b30db5cd82525ab940ab08c51b2f90b12959124ea0e544000
 74.49 MiB / 74.49 MiB [===================================================] 11s
Copying blob sha256:0ff8f8e34aa61061cf06215d1a1ccaea54209289fe0460436e2d16a303cc638e
 126 B / 126 B [============================================================] 0s
Copying config sha256:bd36346540f32240a02694972baf7586a479aaf51796352e92ec9e0c462b996c
 3.71 KiB / 3.71 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: MAINTAINER tom tsweeney@redhat.com
STEP 3: WORKDIR /go/src/component-docker
STEP 4: COPY ./ /go/src/component-docker
STEP 5: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.override_kernel_check=true]localhost/tom:latest
Getting image source signatures
Skipping fetch of repeat blob sha256:df64d3292fd6194b7865d7326af5255db6d81e9df29f48adde61a918fbd8c332
Skipping fetch of repeat blob sha256:c54f8a17910a27d799d33dfd0997625eb72ea10682dd2c033732d0c775e0589e
Skipping fetch of repeat blob sha256:93448d8c26050418b6a9ef671217db21b8063c4ee2319888f2d76600e433736a
Skipping fetch of repeat blob sha256:cc5fec2c1edcbc1d52a40e30c851397a4d54dbdc9c9ca3f2dab75cdaa60c5fbe
Skipping fetch of repeat blob sha256:ac1c7fa88ed0ce3294d58e555b739aca338764ccca6852bd17d6968035b5e514
Copying blob sha256:6141dc279fe7ba371e3550fd72f4bd78308f78dbe7a3c09672e5378b7ca0fb5a
 41.39 MiB / 41.39 MiB [====================================================] 1s
Copying config sha256:718b456c4b757cd80f6353b51081792b3942e008a169df29fe3d388c7c3d8554
 2.98 KiB / 2.98 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
--> 718b456c4b757cd80f6353b51081792b3942e008a169df29fe3d388c7c3d8554

```

**Comment by coderwangke** [comment]
@rhatdan @TomSweeneyRedHat 
I further tested and verified my guess by running with --debug. When there is a problem with from, it does not return, but when the copy is executed, the error is returned. The reason for the failure of from is due to the network speed.
# Limit network speed
sudo wondershaper eth0 5 5

# buildah --debug bud -f /root/src/buildah-test-dockerfile/Dockerfile -t test:latest /root/src/buildah-test-dockerfile/
DEBU[0000] [graphdriver] trying provided driver "overlay"
DEBU[0000] overlay: override_kernelcheck=true
DEBU[0000] overlay test mount with multiple lowers succeeded
DEBU[0000] backingFs=extfs, projectQuotaSupported=false, useNativeDiff=true
DEBU[0000] reading local Dockerfile "/root/src/buildah-test-dockerfile/Dockerfile"
DEBU[0000] FROM "golang:1.10-alpine"
STEP 1: FROM golang:1.10-alpine
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]localhost/golang:1.10-alpine"
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]localhost/golang:1.10-alpine"
DEBU[0000] reference "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]localhost/golang:1.10-alpine" does not resolve to an image ID
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]localhost/golang:1.10-alpine"
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]localhost/golang:1.10-alpine"
DEBU[0000] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0000]  Using "default-docker" configuration
DEBU[0000]  No signature storage configuration found for localhost/golang:1.10-alpine
DEBU[0000] Looking for TLS certificates and private keys in /etc/docker/certs.d/localhost
DEBU[0000] copying "docker://localhost/golang:1.10-alpine" to "localhost/golang:1.10-alpine"
DEBU[0000] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0000]  Using "default-docker" configuration
DEBU[0000]  No signature storage configuration found for localhost/golang:1.10-alpine
DEBU[0000] Looking for TLS certificates and private keys in /etc/docker/certs.d/localhost
DEBU[0000] GET https://localhost/v2/
DEBU[0000] Ping https://localhost/v2/ err &url.Error{Op:"Get", URL:"https://localhost/v2/", Err:(*net.OpError)(0xc4202c0690)}
DEBU[0000] GET https://localhost/v1/_ping
DEBU[0000] Ping https://localhost/v1/_ping err &url.Error{Op:"Get", URL:"https://localhost/v1/_ping", Err:(*net.OpError)(0xc4202c0960)}
DEBU[0000] error pulling image "localhost/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0000] unable to pull and read image "localhost/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]docker.io/library/golang:1.10-alpine"
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]docker.io/library/golang:1.10-alpine"
DEBU[0000] reference "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]docker.io/library/golang:1.10-alpine" does not resolve to an image ID
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]docker.io/library/golang:1.10-alpine"
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]docker.io/library/golang:1.10-alpine"
DEBU[0000] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0000]  Using "default-docker" configuration
DEBU[0000]  No signature storage configuration found for docker.io/library/golang:1.10-alpine
DEBU[0000] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io
DEBU[0000] copying "docker://docker.io/library/golang:1.10-alpine" to "docker.io/library/golang:1.10-alpine"
DEBU[0000] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0000]  Using "default-docker" configuration
DEBU[0000]  No signature storage configuration found for docker.io/library/golang:1.10-alpine
DEBU[0000] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io
DEBU[0000] GET https://registry-1.docker.io/v2/
DEBU[0018] Ping https://registry-1.docker.io/v2/ err &url.Error{Op:"Get", URL:"https://registry-1.docker.io/v2/", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0018] GET https://registry-1.docker.io/v1/_ping
DEBU[0032] Ping https://registry-1.docker.io/v1/_ping err &url.Error{Op:"Get", URL:"https://registry-1.docker.io/v1/_ping", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0032] error pulling image "docker.io/library/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0032] unable to pull and read image "docker.io/library/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0032] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.fedoraproject.org/golang:1.10-alpine"
DEBU[0032] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.fedoraproject.org/golang:1.10-alpine"
DEBU[0032] reference "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.fedoraproject.org/golang:1.10-alpine" does not resolve to an image ID
DEBU[0032] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.fedoraproject.org/golang:1.10-alpine"
DEBU[0032] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.fedoraproject.org/golang:1.10-alpine"
DEBU[0032] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0032]  Using "default-docker" configuration
DEBU[0032]  No signature storage configuration found for registry.fedoraproject.org/golang:1.10-alpine
DEBU[0032] Looking for TLS certificates and private keys in /etc/docker/certs.d/registry.fedoraproject.org
DEBU[0032] copying "docker://registry.fedoraproject.org/golang:1.10-alpine" to "registry.fedoraproject.org/golang:1.10-alpine"
DEBU[0032] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0032]  Using "default-docker" configuration
DEBU[0032]  No signature storage configuration found for registry.fedoraproject.org/golang:1.10-alpine
DEBU[0032] Looking for TLS certificates and private keys in /etc/docker/certs.d/registry.fedoraproject.org
DEBU[0032] GET https://registry.fedoraproject.org/v2/
DEBU[0043] Ping https://registry.fedoraproject.org/v2/ err &url.Error{Op:"Get", URL:"https://registry.fedoraproject.org/v2/", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0043] GET https://registry.fedoraproject.org/v1/_ping
DEBU[0056] Ping https://registry.fedoraproject.org/v1/_ping err &url.Error{Op:"Get", URL:"https://registry.fedoraproject.org/v1/_ping", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0056] error pulling image "registry.fedoraproject.org/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0056] unable to pull and read image "registry.fedoraproject.org/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0056] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]quay.io/golang:1.10-alpine"
DEBU[0056] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]quay.io/golang:1.10-alpine"
DEBU[0056] reference "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]quay.io/golang:1.10-alpine" does not resolve to an image ID
DEBU[0056] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]quay.io/golang:1.10-alpine"
DEBU[0056] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]quay.io/golang:1.10-alpine"
DEBU[0056] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0056]  Using "default-docker" configuration
DEBU[0056]  No signature storage configuration found for quay.io/golang:1.10-alpine
DEBU[0056] Looking for TLS certificates and private keys in /etc/docker/certs.d/quay.io
DEBU[0056] copying "docker://quay.io/golang:1.10-alpine" to "quay.io/golang:1.10-alpine"
DEBU[0056] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0056]  Using "default-docker" configuration
DEBU[0056]  No signature storage configuration found for quay.io/golang:1.10-alpine
DEBU[0056] Looking for TLS certificates and private keys in /etc/docker/certs.d/quay.io
DEBU[0056] GET https://quay.io/v2/
DEBU[0066] Ping https://quay.io/v2/ err &url.Error{Op:"Get", URL:"https://quay.io/v2/", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0066] GET https://quay.io/v1/_ping
DEBU[0080] Ping https://quay.io/v1/_ping err &url.Error{Op:"Get", URL:"https://quay.io/v1/_ping", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0080] error pulling image "quay.io/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0080] unable to pull and read image "quay.io/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0080] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.access.redhat.com/golang:1.10-alpine"
DEBU[0080] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.access.redhat.com/golang:1.10-alpine"
DEBU[0080] reference "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.access.redhat.com/golang:1.10-alpine" does not resolve to an image ID
DEBU[0080] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.access.redhat.com/golang:1.10-alpine"
DEBU[0080] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.access.redhat.com/golang:1.10-alpine"
DEBU[0080] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0080]  Using "default-docker" configuration
DEBU[0080]  No signature storage configuration found for registry.access.redhat.com/golang:1.10-alpine
DEBU[0080] Looking for TLS certificates and private keys in /etc/docker/certs.d/registry.access.redhat.com
DEBU[0080] copying "docker://registry.access.redhat.com/golang:1.10-alpine" to "registry.access.redhat.com/golang:1.10-alpine"
DEBU[0080] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0080]  Using "default-docker" configuration
DEBU[0080]  No signature storage configuration found for registry.access.redhat.com/golang:1.10-alpine
DEBU[0080] Looking for TLS certificates and private keys in /etc/docker/certs.d/registry.access.redhat.com
DEBU[0080] GET https://registry.access.redhat.com/v2/
DEBU[0091] Ping https://registry.access.redhat.com/v2/ err &url.Error{Op:"Get", URL:"https://registry.access.redhat.com/v2/", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0091] GET https://registry.access.redhat.com/v1/_ping
DEBU[0104] Ping https://registry.access.redhat.com/v1/_ping err &url.Error{Op:"Get", URL:"https://registry.access.redhat.com/v1/_ping", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0104] error pulling image "registry.access.redhat.com/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0104] unable to pull and read image "registry.access.redhat.com/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0104] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.centos.org/golang:1.10-alpine"
DEBU[0104] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.centos.org/golang:1.10-alpine"
DEBU[0104] reference "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.centos.org/golang:1.10-alpine" does not resolve to an image ID
DEBU[0104] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.centos.org/golang:1.10-alpine"
DEBU[0104] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]registry.centos.org/golang:1.10-alpine"
DEBU[0104] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0104]  Using "default-docker" configuration
DEBU[0104]  No signature storage configuration found for registry.centos.org/golang:1.10-alpine
DEBU[0104] Looking for TLS certificates and private keys in /etc/docker/certs.d/registry.centos.org
DEBU[0104] copying "docker://registry.centos.org/golang:1.10-alpine" to "registry.centos.org/golang:1.10-alpine"
DEBU[0104] Using registries.d directory /etc/containers/registries.d for sigstore configuration
DEBU[0104]  Using "default-docker" configuration
DEBU[0104]  No signature storage configuration found for registry.centos.org/golang:1.10-alpine
DEBU[0104] Looking for TLS certificates and private keys in /etc/docker/certs.d/registry.centos.org
DEBU[0104] GET https://registry.centos.org/v2/
DEBU[0115] Ping https://registry.centos.org/v2/ err &url.Error{Op:"Get", URL:"https://registry.centos.org/v2/", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0115] GET https://registry.centos.org/v1/_ping
DEBU[0137] Ping https://registry.centos.org/v1/_ping err &url.Error{Op:"Get", URL:"https://registry.centos.org/v1/_ping", Err:http.tlsHandshakeTimeoutError{}}
DEBU[0137] error pulling image "registry.centos.org/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0137] unable to pull and read image "registry.centos.org/golang:1.10-alpine": unable to find image in the registries defined in "/etc/containers/registries.conf"
DEBU[0137] Parsed Step: {Env:[] Command:maintainer Args:[halewang <halewang@tencent.com>] Flags:[] Attrs:map[] Message:MAINTAINER halewang <halewang@tencent.com> Original:MAINTAINER halewang <halewang@tencent.com>}
STEP 2: MAINTAINER halewang <halewang@tencent.com>
DEBU[0137] Parsed Step: {Env:[] Command:workdir Args:[/go/src/component-docker] Flags:[] Attrs:map[] Message:WORKDIR /go/src/component-docker Original:WORKDIR /go/src/component-docker}
STEP 3: WORKDIR /go/src/component-docker
DEBU[0137] Parsed Step: {Env:[] Command:copy Args:[./ /go/src/component-docker] Flags:[] Attrs:map[] Message:COPY ./ /go/src/component-docker Original:COPY ./ /go/src/component-docker}
STEP 4: COPY ./ /go/src/component-docker
DEBU[0137] COPY []string(nil), imagebuilder.Copy{FromFS:false, From:"", Src:[]string{"./"}, Dest:"/go/src/component-docker", Download:false}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0xa2b835]

goroutine 1 [running]:
github.com/projectatomic/buildah.(*Builder).Add(0x0, 0xc420364180, 0x18, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc420152de0, 0x1, ...)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/_build/src/github.com/projectatomic/buildah/add.go:87 +0x45
github.com/projectatomic/buildah/imagebuildah.(*Executor).Copy(0xc4200df600, 0x0, 0x0, 0x0, 0xc4203fb450, 0x1, 0x1, 0x0, 0xc4201d26c0)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/_build/src/github.com/projectatomic/buildah/imagebuildah/build.go:446 +0x616
github.com/openshift/imagebuilder.(*Builder).Run(0xc4200deb00, 0xc42030ae80, 0xcfc840, 0xc4200df600, 0x0, 0x0, 0x0)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/src/github.com/openshift/imagebuilder/builder.go:284 +0x341
github.com/projectatomic/buildah/imagebuildah.(*Executor).Execute(0xc4200df600, 0xcfc580, 0xc420024180, 0xc4200deb00, 0xc42009ff10, 0xc420360005, 0x12)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/_build/src/github.com/projectatomic/buildah/imagebuildah/build.go:754 +0xc4b
github.com/projectatomic/buildah/imagebuildah.(*Executor).Build(0xc4200de000, 0xcfc580, 0xc420024180, 0xc42008d7c0, 0x2, 0x2, 0x0, 0x0)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/_build/src/github.com/projectatomic/buildah/imagebuildah/build.go:1141 +0x247
github.com/projectatomic/buildah/imagebuildah.BuildDockerfiles(0xcfc580, 0xc420024180, 0xd0b960, 0xc4200b26c0, 0x7ffe61b89ed4, 0x21, 0x0, 0x0, 0x0, 0x0, ...)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/_build/src/github.com/projectatomic/buildah/imagebuildah/build.go:1256 +0xfe7
main.budCmd(0xc4200dc160, 0x0, 0x0)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/cmd/buildah/bud.go:239 +0x1611
github.com/urfave/cli.HandleAction(0xb4b6e0, 0xc8f108, 0xc4200dc160, 0x0, 0xc42009c060)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/src/github.com/urfave/cli/app.go:501 +0xc8
github.com/urfave/cli.Command.Run(0xc5d1e0, 0x16, 0x0, 0x0, 0x123b130, 0x1, 0x1, 0xc781b3, 0x31, 0x0, ...)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/src/github.com/urfave/cli/command.go:165 +0x47d
github.com/urfave/cli.(*App).Run(0xc420460380, 0xc42001e080, 0x8, 0x8, 0x0, 0x0)
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/src/github.com/urfave/cli/app.go:259 +0x6e8
main.main()
	/builddir/build/BUILD/buildah-4888163cf12b4e0c0d32ef7eb0ed941c2c81d4f7/cmd/buildah/main.go:120 +0xe22

**Comment by rhatdan** [comment]
Your theory is that the from command actually fails, but does not stop the Dockerfile processing, and eventually the copy command blows up?

@mtrmac @nalind PTAL

**Comment by TomSweeneyRedHat** [comment]
@coderwangke Can you upgrade to v1.4 or later?  We'd a number of error handling fixes that went into v1.4 aht might have addressed this that weren't in v1.3.

**Comment by mtrmac** [comment]
> @coderwangke Can you upgrade to v1.4 or later? We'd a number of error handling fixes that went into v1.4 that might have addressed this that weren't in v1.3.

Yes, this was almost certainly fixed by the second commit of #947.

**Comment by coderwangke** [comment]
thanks @TomSweeneyRedHat @rhatdan 
After I upgraded to v1.4, from command returned an error.
